iroh-net/src/hp/magicsock/conn.rs

```diff
@@ -2293,6 +2299,61 @@ fn log_endpoint_change(endpoints: &[cfg::Endpoint]) {
     });
 }
 
+/// Addresses to which to which we can send. This is either a UDP or a derp address.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub enum SendAddr {
+    /// UDP, the ip addr.
+    Udp(SocketAddr),
+    /// Derp, region id.
+    Derp(u16),
```

Comment:
I'm pretty tempted to also newtype the `u16` into a `DerpRegion`, not really important here but for the functions that take it as a parameter or structs that store the region id that would probably also make things nicer.

Reply:
I thought about it, but wanted to avoid too many types for the moment 😅